src/plone/z3cform/traversal.py

```diff
@@ -71,7 +72,7 @@ def traverse(self, name, ignored):
                     # part should be integer index in list, look it up
                     target = target.widgets[int(part)]
                 except IndexError:
-                    raise TraversalError("'" + part + "' not in range")
+                    target = target.getWidget(int(part))
```

Comment:
We could limit this to check if the integer is `len(target.widgets)`, but I do not see any issue in allowing the traverser to render the widget with any index